SignalR.Npm.FunctionalTests.npmproj fail on Fedora 37+

This test project fails when I run it on Fedora. At the end of the build log I see this:
```
Build FAILED.

(node : 134897) error : Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification. [/home/tmds/repos/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
(node : 134897) error : Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification. [/home/tmds/repos/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
(node : 134898) error : Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification. [/home/tmds/repos/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
(node : 134898) error : Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification. [/home/tmds/repos/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
*** SPEC STARTED : connection with FetchHttpClient contains server response in error [2023-05-03T07: 48:22.050Z] *** [/home/tmds/repos/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
*** SPEC DONE : connection with FetchHttpClient contains server response in error [2023-05-03T07: 48:22.059Z; Duration: 9ms] *** [/home/tmds/repos/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
/home/tmds/repos/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj(24,5): error : Command failed with exit code 1.
/home/tmds/repos/aspnetcore/.packages/microsoft.dotnet.arcade.sdk/8.0.0-beta.23226.4/tools/Build.proj(267,5): error MSB4181: The "MSBuild" task returned false but did not log an error.

```

Because of the `NODE_TLS_REJECT_UNAUTHORIZED` mention, it could be related to:
https://github.com/dotnet/aspnetcore/blob/9c38b3749a69050e0670b8750d9828cf47a19b41/src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts#L216-L229

As part of the test log I also see:
```
FAIL ts/HubConnectionTests.ts (8.189 s)
...
  Test Suites: 1 failed, 3 passed, 4 total
  Tests:       3 failed, 5 skipped, 152 passed, 160 total
```

This is the full test log: [test.log](https://github.com/dotnet/aspnetcore/files/11380556/test.log)

@BrennanConroy can you help me figure out what the issue is?

--- a/src/SignalR/clients/ts/signalr/src/FetchHttpClient.ts
+++ b/src/SignalR/clients/ts/signalr/src/FetchHttpClient.ts
@@ -20,14 +20,21 @@ export class FetchHttpClient extends HttpClient {
         super();
         this._logger = logger;
 
-        if (typeof fetch === "undefined") {
+        // Node added a fetch implementation to the global scope starting in v18.
+        // We need to add a cookie jar in node to be able to share cookies with WebSocket
+        if (typeof fetch === "undefined" || Platform.isNode) {
             // In order to ignore the dynamic require in webpack builds we need to do this magic
             // @ts-ignore: TS doesn't know about these names
             const requireFunc = typeof __webpack_require__ === "function" ? __non_webpack_require__ : require;
 
             // Cookies aren't automatically handled in Node so we need to add a CookieJar to preserve cookies across requests
             this._jar = new (requireFunc("tough-cookie")).CookieJar();
-            this._fetchType = requireFunc("node-fetch");
+
+            // Use fetch from Node if available
+            this._fetchType = fetch;
+            if (typeof fetch === "undefined") {
+                this._fetchType = requireFunc("node-fetch");
+            }
 
             // node-fetch doesn't have a nice API for getting and setting cookies
             // fetch-cookie will wrap a fetch implementation with a default CookieJar or a provided one

--- a/src/SignalR/clients/ts/signalr/src/FetchHttpClient.ts
+++ b/src/SignalR/clients/ts/signalr/src/FetchHttpClient.ts
@@ -30,10 +30,11 @@ export class FetchHttpClient extends HttpClient {
             // Cookies aren't automatically handled in Node so we need to add a CookieJar to preserve cookies across requests
             this._jar = new (requireFunc("tough-cookie")).CookieJar();
 
-            // Use fetch from Node if available
-            this._fetchType = fetch;
             if (typeof fetch === "undefined") {
                 this._fetchType = requireFunc("node-fetch");
+            } else {
+                // Use fetch from Node if available
+                this._fetchType = fetch;
             }
 
             // node-fetch doesn't have a nice API for getting and setting cookies

